hail/python/hail/table.py

```diff
@@ -1317,8 +1338,43 @@ def format_line(values, widths, right_align):
 
         return s
 
-    @typecheck_method(n=int, width=int, truncate=nullable(int), types=bool, handler=anyfunc)
-    def show(self, n=10, width=90, truncate=None, types=True, handler=print):
+    def _html_str(self, n, types):
+        import cgi
+
+        def hl_format(v):
+            return cgi.escape(Table._hl_repr(v))
+
+        t = self
+        t = t.flatten()
+        fields = list(t.row)
+
+        formatted_t = t.select(**{k: hl_format(v) for (k, v) in t.row.items()})
```

Comment:
This won't work - it'll try to call hl_format on expressions. hl_format needs to be called on strings after the `take` localizes them to python

Reply:
wait, but isn't this calling `cgi.escape` inside of a TableMapRows (select) above? How does that work?

Reply:
lmao nice: https://github.com/python/cpython/blob/3.5/Lib/html/__init__.py#L12

replace just works.

Reply:
```python
import html
print(html.escape(hl.str("tlocus<GRCh37>"))._ir)
```
```
(Apply replace (Apply replace (Apply replace (Apply replace (Apply replace (Str "tlocus<GRCh37>") (Str "&") (Str "&amp;")) (Str "<") (Str "&lt;")) (Str ">") (Str "&gt;")) (Str "\"") (Str "&quot;")) (Str "'") (Str "&#x27;"))
```

Reply:
I don't really like this -- I think there's nothing that prevents the `cgi` or `html` implementation from changing and breaking this, and Hail string manipulation is also *extremely* slow right now.

Can we just call `hl_format` on each string coming out in line 1367?